File: Pset 6/greedy.py
#!/usr/bin/env python3

import cs50

def main():
    
    # ask user to input amount of change required
    while True:
        print("O hai! How much change is owed? ", end="")
        change = cs50.get_float()
        if change > 0:
            break
        
    # convert change to cents
    cents = round(change * 100)
    
    # calculate number of quarters
    quarters = cents // 25
    remaining = cents % 25
    
    # calculate number of dimes
    dimes = remaining // 10
    remaining = remaining % 10
    
    # calculate number of nickels
    nickels = remaining // 5
    remaining = remaining % 5
    
    # calculate number of coins required to make change
    coin_count = quarters + dimes + nickels + remaining
    
    print(coin_count)
    
if __name__ == "__main__":
    main()

File: Pset 1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float change = 0;

int main(void)
{
    // ask user to input amount of change required
    do
    {
        printf("O hai! How much change is owed?\n");
        change = get_float();
    }
    while (change <= 0);
    
    // convert change to cents
    int cents = (int)round(change * 100);
    
    // calculate number of quarters
    int quarters = cents / 25;
    int remaining = cents % 25;
    
    // calculate number of dimes
    int dimes = remaining / 10;
    remaining = remaining % 10;
    
    // calculate number of nickels
    int nickels = remaining / 5;
    remaining = remaining % 5;
    
    // remaining at this point is number of pennies
    
    // calculate number of coins required to make change
    int coin_count = quarters + dimes + nickels + remaining;
    
    printf("%i\n", coin_count);
}